### ISSUE TITLE
Is lazy loaded module's promise resolved before its dependencies are loaded and resolved?

### ISSUE THREAD

**Issue by nkanand4** [ocombe/ocLazyLoad#262]
I have a module called "diagnostics", which is being lazy loaded. In that module's dependency list, I have list of files that this module will need to complete itself, e.g few controllers and directives.
Now, In the **then** function of the module X load, I assume that all the controllers that are defined in modules X's dependency list are also available now. From the observation of the screenshots below I realize that my understanding is wrong. You can notice that "diagnosticsController.js" is not loaded yet. However, my breakpoint in the **then** function is hit.

Main module file:
![diagnostics-js](https://cloud.githubusercontent.com/assets/667996/10774265/25bc8e62-7cce-11e5-911c-16e42a9f965f.PNG)

Network requests which shows that diagnosticsController.js is still pending.
![network-requests](https://cloud.githubusercontent.com/assets/667996/10774280/3169eb42-7cce-11e5-8d4d-7378816d288b.PNG)

**Comment by nkanand4** [comment]
I forgot to mention this happened only sometime when I did a "Shift + refresh button" and now its not happening at all. Could this be similar to https://github.com/ocombe/ocLazyLoad/issues/259?
Anyways, restarting the browser all afresh seems to have resolved the issue for now. Also, I looked at the code and it seems the oclazyload.load()'s returned promise will not be resolved until all the deps of deps are not resolved. So, it should work as I was expecting it to work.
